Source/WebCore/Modules/mediasession/MediaSession.cpp

```diff
@@ -237,6 +237,7 @@ void MediaSession::setPlaybackState(MediaSessionPlaybackState state)
 
 void MediaSession::setActionHandler(MediaSessionAction action, RefPtr<MediaSessionActionHandler>&& handler)
 {
+    Locker lock { m_actionHandlersLock };
```

Comment:
We should consider reducing the scope of the lock to just the HashMap operations.